templates/adding_a_new_example_script/{{cookiecutter.directory_name}}/run_{{cookiecutter.example_shortcut}}.py

```diff
@@ -226,15 +226,15 @@ def main():
         datefmt="%m/%d/%Y %H:%M:%S",
         handlers=[logging.StreamHandler(sys.stdout)],
     )
-    logger.setLevel(logging.INFO if is_main_process(training_args.local_rank) else logging.WARN)
+    logger.setLevel(logging.INFO if training_args.should_log else logging.WARN)
```

Comment:
As we have lots of warnings this means we still end up with many dozens of copies of warnings on multinode.

Perhaps `log_on_each_node` shouldn't be bool, but instead have the level to set for multinode, so if I set it:
```
log_on_each_node=error
```
then only error logs are replicated on all nodes.

Reply:
That's a different API, and this controls what happens for each process different than local main (or main main is the flag is set to False). We can adjust the level of logging on each child process to Error in another PR perhaps?

Reply:
Yes, of course. Ideally it should be user-configurable - perhaps someone does want WARN to be logged on all nodes.